cmake/CMakeLists.txt

```diff
@@ -227,7 +227,15 @@ pkg_depends(USER-LB MPI)
 pkg_depends(USER-PHONON KSPACE)
 pkg_depends(USER-SCAFACOS MPI)
 
+# detect if we may enable OpenMP support by default
+set(BUILD_OMP_DEFAULT OFF)
 find_package(OpenMP QUIET)
+if(OpenMP_FOUND)
+  check_include_file_cxx(omp.h HAVE_OMP_H_INCLUDE)
```

Comment:
I am still very confused why this check of `omp.h` is needed? The existence of `omp.h` is already check by a `try_compile` in [FindOpenMP.cmake](https://github.com/Kitware/CMake/blob/v3.10.0/Modules/FindOpenMP.cmake#L306).

@robertmaynard @bradking?

Reply:
[CMake MR 3916](https://gitlab.kitware.com/cmake/cmake/-/merge_requests/3916), included in CMake 3.16, added detection of `OpenMP_<lang>_INCLUDE_DIRS` for the `omp.h` location.  See issues linked from that MR for details on problems that occurred before that fix.

Reply:
I tried on multiple machines (e.g. Fedora using clang without libomp-devel or MacOSX with Xcode) and `OpenMP_FOUND` is not sufficient as it is true even if `omp.h` is missing.

Reply:
@akohlmey thanks.  Ideally `omp.h` should be in the compiler's default include search path, at least when omp flags are used, but there are some cases where it is not.  The CMake MR I linked fixes at least one such case.  I don't know if there are others.

We should try to get this fixed in upstream CMake's FindOpenMP if it isn't already.  Please try CMake 3.17.  If you can still produce a case where `OpenMP_FOUND` is true but `OpenMP_<lang>_INCLUDE_DIRS` is not sufficient to include `omp.h` then please open an issue in the [CMake Issue Tracker](https://gitlab.kitware.com/cmake/cmake/issues) to describe the platform/compiler combination where it fails.

Reply:
@bradking as already mentioned, the two cases I was describing are where `omp.h` is not even installed (or the corresponding runtime library) by default with the compiler (clang), while the compiler does accept the flag enabling OpenMP. 
Since our current goal for LAMMPS is compatibility with older CMake versions down to 3.10 (to support older Linux variants out of the box) we need to stick with the workaround in any case.